subcommand/server-acl-init/command.go

```diff
@@ -186,6 +211,21 @@ func (c *Command) Run(args []string) int {
 		aclReplicationToken = strings.TrimSpace(string(tokenBytes))
 	}
 
+	var providedBootstrapToken string
+	if c.flagBootstrapTokenFile != "" {
```

Comment:
It seemed simpler to me to have only one flag for the file rather than having two flags - for secret name and secret key. I don't think reading it from a file adds an extra hop - it just replaces one API call with another. Plus, if reading from API, we'd still have to do similar error checks: 1) fail if secret doesn't exist; and 2) fail if the value of the secret key is empty. So it will likely not reduce the number of lines either.

Philosophically, it also seems appropriate for this job to only read/write secrets that it creates/manages. So this implementation follows the existing pattern of reading the CA cert and ACL replication token from a file.
 
We should probably also look into restricting the role rules for this job in the Helm chart to only read the list of secrets we expect. It would probably be cumbersome, but better from a security perspective.

Reply:
My first instinct with anything running in a container that needs files as input is that it's often awkward to be able to provide it. That's probably not an issue here because it's managed in the helm chart and it's straightforward to mount the secret as a file.

This process does create and manage the bootstrapping secret, just not in this particular configuration. Given that, one could argue that it's confusing for the job to sometimes access the secret directly and other times not.

Reply:
Yeah, that's true. It makes it inconsistent now, where previously it was only using the API for Kube secrets. There is a pattern though: secrets that this job doesn't create/manage, we provide via files (e.g. consul ca cert, acl replication token, bootstrap token), and secrets that this job creates and updates it manages via the API.

I do like also that it decouples us from Kube secrets a little bit.